### PR TITLE
Added ability to bulk select, pause & resume

### PR DESCRIPTION
**Pull request by simshaun**
# Description

Introduces the ability to pause & resume multiple monitors.

## Type of change

- User interface (UI)
- New feature (non-breaking change which adds functionality)
- Translation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

image

image

On tablet (~780px width):
image

image

image

### CODE REVIEW

File: src/components/MonitorList.vue [hunk 1]
```diff
@@ -129,6 +156,15 @@ export default {
 
             return result;
         },
+
+        isDarkTheme() {
```
Comment (simshaun) [inline src/components/MonitorList.vue:160]: Not particularly happy about having this here. Wanted to import the theme mixin, but it contains side-effects.

File: src/components/MonitorList.vue [hunk 2]
```diff
@@ -156,7 +192,51 @@ export default {
...
+        /** Toggle selection of monitor */
+        toggleSelection(id) {
+            if (this.isSelected(id)) {
+                delete this.selectedMonitors[id];
+            } else {
+                this.selectedMonitors[id] = true;
+            }
+        },
+        /** Determine if monitor is selected */
+        isSelected(id) {
```
[the remark this answers is not in the document]
Comment (simshaun) [reply src/components/MonitorList.vue:219]: added

**Comment by GrantBirki** [comment]
@simshaun This looks awesome! One minor critique, could you scale down the size of the ⏯️ (play/pause) buttons a smidgen? They seem slightly larger than other button elements on that page but perhaps it is just me

**Comment by simshaun** [comment]
> @simshaun This looks awesome! One minor critique, could you scale down the size of the ⏯️ (play/pause) buttons a smidgen? They seem slightly larger than other button elements on that page but perhaps it is just me

I just used the same icon (without changing size) as other elements I found in the code, but I agree they look too big.

I changed it so now they look like this:

image

**Comment by louislam** [comment]
Need volunteers to test this pull request.

You can test this pull request by using the following docker command:

```bash
docker run --rm -it -p 3000:3000 -p 3001:3001 --pull always -e 'UPTIME_KUMA_GH_REPO=simshaun:multi-select' louislam/uptime-kuma:pr-test
```

Go to http://localhost:3000 after started.

**Comment by YehowahLiu** [comment]
> Need volunteers to test this pull request.
> 
> You can test this pull request by using the following docker command:
> 
> ```shell
> docker run --rm -it -p 3000:3000 -p 3001:3001 --pull always -e 'UPTIME_KUMA_GH_REPO=simshaun:multi-select' louislam/uptime-kuma:pr-test
> ```
> 
> Go to http://localhost:3000 after started.

Tested, LGTM

**Comment by chakflying** [comment]
`@click.stop` doesn't work on firefox for some reason.
https://user-images.githubusercontent.com/3271800/210354071-0176df5f-56ce-44f5-bb4e-427d1002c9d3.mp4

**Comment by simshaun** [comment]
I've rebased and got things working again:

image

image

image

**Comment by chakflying** [comment]
- Not a fan of the default styles provided by "dropdown-menu-dark". We have already handled this in Header's Dropdown Menu (`Layout.vue`).
- "Cancel" doesn't really make sense inside the "With Selected..." dropdown. I understand it's an issue of width, my idea was to permanently have an icon at the top left to toggle on/off the "Select Mode". See if you're interested.

image

**Comment by louislam** [comment]
Since #3312 and this pr both are modifying the same place, I think there should be some code conflicts. I will try to merge #3321 first and then this.

**Comment by louislam** [comment]
Merged, it seems fine.

I will try to update the style to match with #3312's style. Also I may need to change back the monitor list item's padding.

Thanks.

image

**Comment by louislam** [comment]
Trying to update the css to match the filter style and I expanded the `With Selected...` menu into a new row below the filter.

Also, I found that those flex css causes a lot of wired bugs to the monitor list, I removed them all and use `float left` to position the checkbox. 

The heartbeat bars are actually overflow after the checkbox are shown, it can be fixed by re-render the heartbeat bar, but the performance is really bad, so I think it is acceptable to keep like this first.

image
image

**Comment by jesperordrup** [comment]
Does this really close #1457 ? I don't see any UI that adds this. I see some examples of how to work with existing by multiselecting

**Pull request merged**